resources/lang/vi/message.php

```diff
@@ -1,5 +1,6 @@
 <?php
 return [
-    'welcome' => 'Chào mừng bạn đến với Website!',
-    'successfully' => "Thành công"
+    'Xin chào' => 'Chào mừng bạn đến với Website!',
+    'Thành công' => "Thành công",
+    'Thất bại' => 'thất bại'
```

Comment:
Hiểu sai ý anh rồi em ơi, nó nên là `xin_chao => 'Chào mừng bạn đến với Website!'`

Reply:
Cảnh ơi. Key  nên là Xin chào thay vì xin_chao

Vì nếu như từ đó không tồn tại thì nó sẽ hiện thị cái key. nên là key cũng nên rõ 1 chút.

Reply:
Cái này nếu làm thế sẽ có những vấn đề sau:
- Codeding convention sẽ không thống nhất lúc thì `xin_chao`, lúc thì Xin Chào làm key ...
- Nếu web này làm đa ngôn ngữ chẳng hạn thì lúc gọi ra sẽ là `__('lang.Xin Chào')` cái này anh không chắc sẽ hoạt động được và kể cả có hoạt động thì nhìn thôi cũng đã thấy không hợp lệ 
=> Cái này mình hoàn toàn có thể sử dụng custom Rule để làm được 1 message hoàn chỉnh, hoặc cách đơn giản nhất là trả về `Giá trị không hợp lệ`

Reply:
>Codeding convention sẽ không thống nhất lúc thì xin_chao, lúc thì Xin Chào làm key ...
Sao lại khong thông nhất. đã để key là Xin Chao thì tất cả khi sử dụng sẽ là Xin Chào, sẽ không có dùng cái xin_chao nữa

>Nếu web này làm đa ngôn ngữ chẳng hạn thì lúc gọi ra sẽ là __('lang.Xin Chào') cái này anh không chắc sẽ hoạt động được và kể cả có hoạt động thì nhìn thôi cũng đã thấy không hợp lệ

Sử dụng `Lang` để đọc thì nó đúng là nhìn không thuận mặt chứ không phải không hợp lệ.

Về mặt hoạt động cả cái đa ngôn ngữ này thì nếu key không tồn tại thì nó sẽ in Key đó ra. nên nếu để xin_chao như này, trường hợp k tồn tại thì màn hình xâu key đó ra thì về mặt người dùng là không được.

Nên tránh tình trạng code nhìn không thuận mắt như cái ở trên. thì đưa file đa ngôn ngữ về dạng Json. sau đó đọc theo cách này thì sẽ ok và ổn thoả cả đôi bên "__('Xin Chao')"

Reply:
https://laravel.com/docs/6.x/localization#using-translation-strings-as-keys

Reply:
@TrungPaul  Ở trên chỉ là 2 option đẻ e chọn. 
Nếu sự dọng __('lang.Xin Chào') thì key nên là `xin_chao`

nếu sử dụng json file sửa thành `Xin Chào`

Link  a đã gắn ở trên e có thể đọc thêm